Support amqp queues

carlossg opened the pull request.

scoheb commented:
Thanks. Would you be able to update the integration tests (using ATH) to test the support of amqp?

carlossg commented:
I've refactored the activemq one so it uses a generic JMS Connection. The only difference for amqp is at connection factory creation time
I've added unit and integration tests. The amqp one is just extending the activemq one, and that way it runs the same scenarios

ggallen commented:
Will revisit this is amqp support is needed in the future.
